src/box/lua/trigger.c

```diff
@@ -74,20 +135,62 @@ luaT_trigger_call(struct lua_State *L)
 	struct event *event = event_get(event_name, false);
 	if (event == NULL)
 		return 0;
-	int top = lua_gettop(L);
-	int narg = top - 1;
+	int nargs = lua_gettop(L) - 1;
+
+	/*
+	 * If the function is called on Lua-stack from fiber storage, create
+	 * new thread because we cannot dump port_lua to the same stack as
+	 * the underlying one, and it's cheaper to leave fiber storage stack
+	 * to triggers - they won't have to create their own stacks then.
+	 *
+	 * If we have no args, port won't be created, so we don't need to
+	 * create a stack.
+	 */
+	struct lua_State *args_L = L;
+	int args_coro_ref = LUA_REFNIL;
+	if (nargs > 0) {
+		if (fiber()->storage.lua.stack == L) {
```

Comment:
Why can they be different?

Reply:
This part was reworked, so the comment is outdated.